matchzoo/preprocessors/units/tokenize.py

```diff
@@ -15,3 +17,104 @@ def transform(self, input_: str) -> list:
         :return tokens: tokenized tokens as a list.
         """
         return nltk.word_tokenize(input_)
+
+
+class ChineseTokenize(Unit):
+    """Process unit for text containing Chinese tokens."""
+
+    def transform(self, input_: str) -> str:
+        """
+        Process input data from raw terms to processed text.
+
+        :param input_: raw textual input.
+
+        :return output: text with a blank between adjacent Chinese tokens.
+        """
+        output = []
+        for char in input_:
+            cp = ord(char)
+            if is_chinese_char(cp):
+                output.append(" ")
+                output.append(char)
+                output.append(" ")
+            else:
+                output.append(char)
+        return "".join(output)
```

Comment:
Do you think it's correct? Let's say you give two Chinese chars: 'a' and 'b'

`output = [' ', 'a', ' ', ' ', 'b', ' ']`

And you join the result, it will be two blanks instead of one blank as described in your `:return:` section.

Reply:
The unit ``ChineseTokenize`` just adds whitespaces around Chinese chars to solve the cases like ``input_ = "abcd"``, so the return type is str. The redundant whitespaces will be cleaned in the next unit ``BasicTokenize`` and I will change the comment as ``return output: text with at least one blank ...``. Maybe there is a proper name for this unit.

Reply:
they why not remove line 40?